module/chunks/chunkVerifier.go

```diff
@@ -272,6 +281,55 @@ func (fcv *ChunkVerifier) verifyTransactionsInContext(context fvm.Context, chunk
 		return nil, chmodels.NewCFMissingRegisterTouch(nil, chIndex, execResID, problematicTx), nil
 	}
 
+	// check basic execution data root fields
+	if chunkDataPack.ExecutionDataRoot.BlockID != chunk.BlockID {
+		return nil, chmodels.NewCFExecutionDataBlockIDMismatch(chunkDataPack.ExecutionDataRoot.BlockID, chunk.BlockID, chIndex, execResID), nil
+	}
+
+	if len(chunkDataPack.ExecutionDataRoot.ChunkExecutionDataIDs) != len(result.Chunks) {
+		return nil, chmodels.NewCFExecutionDataChunksLengthMismatch(len(chunkDataPack.ExecutionDataRoot.ChunkExecutionDataIDs), len(result.Chunks), chIndex, execResID), nil
+	}
+
+	// check our chunk id inside
+	trieUpdate, err := pathfinder.UpdateToTrieUpdate(update, partial.DefaultPathFinderVersion)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to convert to trie update: %w", err)
+	}
+
+	chunkExecutionData := &execution_data.ChunkExecutionData{
+		Collection: chunkDataPack.Collection,
+		Events:     events,
+		TrieUpdate: trieUpdate,
+	}
+
+	cidProvider := provider.NewExecutionDataCIDProvider(execution_data.DefaultSerializer)
```

Comment:
I don't think this method exists in the current version of `provider`

Reply:
its a package method